provider/app.go

```diff
@@ -76,6 +77,39 @@ func appResource() *schema.Resource {
 				ForceNew: true,
 				Optional: true,
 			},
+			"share": {
+				Type: schema.TypeString,
+				Description: "Determines the sharing level of the app. " +
+					"Application sharing is an enterprise feature and any " +
+					"values will be ignored (and sharing disabled) if your " +
+					"deployment is not entitled to use application sharing. " +
+					`Valid values are "owner", "template", "authenticated" ` +
+					`and "public". Level "owner" disables sharing on the ` +
+					"app, so only the workspace owner can access it. Level " +
+					`"template" shares the app with all users that can read ` +
+					`the workspace's template. Level "authenticated" shares ` +
+					`the app with all authenticated users. Level "public" ` +
+					"shares it with any user, including unauthenticated " +
+					"users. Permitted application sharing levels can be " +
+					`controlled via a flag on "coder server". Defaults to ` +
+					`"owner" (sharing disabled).`,
```

Comment:
We should wait to allow the blocking of sharing globally IMO.

Reply:
Talked to Kyle on slack and he agrees we should keep the flag but it should default to all levels being available if you're entitled to the feature.